internal/pkg/diagnostics/diagnostics.go

```diff
@@ -324,23 +324,46 @@ func writeRedacted(errOut, w io.Writer, fullFilePath string, fr client.Diagnosti
 		}
 	}
 
-	_, err := w.Write(*out)
+	_, err := resultWriter.Write(*out)
 	return err
 }
 
-func redactMap(m map[interface{}]interface{}) map[interface{}]interface{} {
-	for k, v := range m {
-		if v != nil && reflect.TypeOf(v).Kind() == reflect.Map {
-			v = redactMap(v.(map[interface{}]interface{}))
-		}
-		if s, ok := k.(string); ok {
-			if redactKey(s) {
-				v = REDACTED
+// Redact sensitive values from the underlying map
+// the whole generic function here is out of paranoia. Although extremely unlikely,
+// we have no way of guaranteeing we'll get a "normal" map[string]interface{},
+// since the diagnostic interface is a bit of a free-for-all
+func redactMap[K comparable](errOut io.Writer, inputMap map[K]interface{}) map[K]interface{} {
```

Comment:
[Suggestion|Go convention-ish]

```suggestion
// redactMap redacts sensitive values from the underlying map
// the whole generic function here is out of paranoia. Although extremely unlikely,
// we have no way of guaranteeing we'll get a "normal" map[string]interface{},
// since the diagnostic interface is a bit of a free-for-all
func redactMap[K comparable](errOut io.Writer, inputMap map[K]interface{}) map[K]interface{} {
```